### PR TITLE
Improve error message when cred helper not found

### PR DESCRIPTION
**Pull request by drigz**
This makes problems like https://github.com/bazelbuild/rules_docker/issues/196 easier to debug.
For me it happened because `puller` was run by Bazel with a restricted
PATH that didn't include `docker-credentials-gcr`. The new error in this
case is:

```
Traceback (most recent call last):
  File "/usr/local/google/home/rodrigoq/.cache/bazel/_bazel_rodrigoq/a2995359558cbf75786e44ea10657e9b/execroot/containerregistry/bazel-out/k8-fastbuild/bin/puller.runfiles/containerregistry/tools/fast_puller_.py", line 97, in <module>
    main()
  File "/usr/local/google/home/rodrigoq/.cache/bazel/_bazel_rodrigoq/a2995359558cbf75786e44ea10657e9b/execroot/containerregistry/bazel-out/k8-fastbuild/bin/puller.runfiles/containerregistry/tools/fast_puller_.py", line 83, in main
    with v2_2_image.FromRegistry(name, creds, transport, accept) as v2_2_img:
  File "/usr/local/google/home/rodrigoq/.cache/bazel/_bazel_rodrigoq/a2995359558cbf75786e44ea10657e9b/execroot/containerregistry/bazel-out/k8-fastbuild/bin/puller.runfiles/containerregistry/client/v2_2/docker_image_.py", line 364, in __enter__
    self._name, self._creds, self._original_transport, docker_http.PULL)
  File "/usr/local/google/home/rodrigoq/.cache/bazel/_bazel_rodrigoq/a2995359558cbf75786e44ea10657e9b/execroot/containerregistry/bazel-out/k8-fastbuild/bin/puller.runfiles/containerregistry/client/v2_2/docker_http_.py", line 198, in __init__
    self._Refresh()
  File "/usr/local/google/home/rodrigoq/.cache/bazel/_bazel_rodrigoq/a2995359558cbf75786e44ea10657e9b/execroot/containerregistry/bazel-out/k8-fastbuild/bin/puller.runfiles/containerregistry/client/v2_2/docker_http_.py", line 285, in _Refresh
    'Authorization': self._basic_creds.Get()
  File "/usr/local/google/home/rodrigoq/.cache/bazel/_bazel_rodrigoq/a2995359558cbf75786e44ea10657e9b/execroot/containerregistry/bazel-out/k8-fastbuild/bin/puller.runfiles/containerregistry/client/docker_creds_.py", line 160, in Get
    raise Exception("executable not found: %s" % bin_name)
Exception: executable not found: docker-credential-gcr
```

### CODE REVIEW

File: client/docker_creds_.py [hunk 1]
```diff
@@ -149,10 +150,16 @@ def Get(self):
...
-                         stdin=subprocess.PIPE,
-                         stderr=subprocess.STDOUT)
+    try:
+      p = subprocess.Popen([bin_name, 'get'],
+                           stdout=subprocess.PIPE,
+                           stdin=subprocess.PIPE,
+                           stderr=subprocess.STDOUT)
+    except OSError as e:
+      if e.errno == errno.ENOENT:
+        raise Exception("executable not found: %s" % bin_name)
```
Comment (dekkagaijin) [inline client/docker_creds_.py]: nit: might as well make this a concatenation

Comment (drigz) [reply client/docker_creds_.py]: Done. Also made the quotes match the surrounding code. Thanks for the review!

**Comment by mattmoor** [comment]
@dekkagaijin We generally import changes into the mono-repo and export to GitHub.  Can you do that here?

**Comment by dekkagaijin** [comment]
Yeah, was in the process of doing that but got distracted last monday. I intimated that process to drigz insternally :p

**Comment by dekkagaijin** [comment]
@mattmoor Actually, wasn't distracted at all. Done.

**Pull request merged**